dataactcore/scripts/deriveHistoricalAssistanceLocation.py

```diff
@@ -0,0 +1,592 @@
+
+
+def fix_fpds_le_state(row, state_by_code):
+    """ Update legal entity state info """
+    # there are several instances where state code is 'nan' in le, which is simply poor storage, clear those out.
+    if row.legal_entity_state_code == 'nan':
+        row.legal_entity_state_code = None
+
+    if not row.legal_entity_state_descrip and row.legal_entity_state_code\
+            and row.legal_entity_state_code.upper() in state_by_code:
+        row.legal_entity_state_descrip = state_by_code[row.legal_entity_state_code.upper()]
+
+
+def fix_fpds_ppop_state(row, state_by_code, state_code_by_fips):
+    """ Update place of performance state info """
+    # fill in the state name where possible
+    if not row.place_of_perfor_state_desc and row.place_of_performance_state:
+        state_code = row.place_of_performance_state.upper()
+        # there are several cases in ppop where state codes are stored as FIPS codes
+        if state_code in state_code_by_fips:
+            state_code = state_code_by_fips[state_code]
+
+        # if the state code (the one given or the one derived above from FIPS) is in the state list, get the name
+        if state_code in state_by_code:
+            row.place_of_perfor_state_desc = state_by_code[state_code]
+
+
+def fix_fpds_le_cd(row):
+    # if the CD is a mashup of CD and state code, get just the last 2 characters (which are the CD)
+    if re.match('.+\d\d$', row.legal_entity_congressional):
+        row.legal_entity_congressional = row.legal_entity_congressional[-2:]
+    # if it's ZZ, just clear it out
+    elif row.legal_entity_congressional == 'ZZ':
+        row.legal_entity_congressional = None
+
+
+def fix_fpds_ppop_cd(row):
+    # if the CD is a mashup of CD and state code, get just the last 2 characters (which are the CD)
+    if re.match('.+\d\d$', row.place_of_performance_congr):
+        row.place_of_performance_congr = row.place_of_performance_congr[-2:]
+    # if it's ZZ, just clear it out
+    elif row.place_of_performance_congr == 'ZZ':
+        row.place_of_performance_congr = None
+
+
+def fix_fpds_le_county(sess, row, county_by_code):
+    """ These are both new columns, so as long as we have the data, we want to try to derive them """
+    # we only want to do the below if we're missing either the county code or name, don't access the zip DB if we don't
+    # have to
+    if not row.legal_entity_county_code or not row.legal_entity_county_name:
+
+        zip_data = get_zip_data(sess, row.legal_entity_zip4)
+        if zip_data:
+            if not row.legal_entity_county_code:
+                row.legal_entity_county_code = zip_data.county_number
+
+            # if we got the zip data and have a state code to work with, if it's valid then grab the county name
+            if not row.legal_entity_county_name and row.legal_entity_state_code:
+                state = row.legal_entity_state_code.upper()
```

Comment:
This is slightly nit-picky but if we're referring to the state code, it'd be more clear for the variable to be named `state_code` so the referenced value is 100% clear.